src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -328,144 +328,580 @@ private void TrySetUnrecoverableError(SyntaxDiagnosticInfo error)
                 ArrayBuilder<Interpolation>? interpolations,
                 out Range closeQuoteRange)
             {
-                kind = _kind;
-                ScanInterpolatedStringLiteralStart(out openQuoteRange);
-                ScanInterpolatedStringLiteralContents(interpolations);
-                ScanInterpolatedStringLiteralEnd(out closeQuoteRange);
+                // Scan through the open-quote portion of this literal, determining important information the rest of
+                // the scanning needs.
+                var start = _lexer.TextWindow.Position;
+                var succeeded = ScanOpenQuote(out kind, out var startingDollarSignCount, out var startingQuoteCount);
+                Debug.Assert(_lexer.TextWindow.Position != start);
+
+                openQuoteRange = start.._lexer.TextWindow.Position;
+                Debug.Assert(kind != default);
```

Comment:
>default [](http://example.com/codeflow?start=37&length=7)

nit: `0` would be more straightforward. Maybe even `> 0`

Reply:
sure :)

Reply:
I've actually remoed this entirely.  at one point the 'kind' was stored as state, and i was worried about someone using the uninitialized state value.  Now, there's no storing.  only computation and returning.  So i'm not concerned about that anymore.